pkg/liqo-controller-manager/resource-request-controller/offerUpdater.go

```diff
@@ -193,15 +197,24 @@ func (u *OfferUpdater) getStorageClasses(ctx context.Context) ([]sharingv1alpha1
 // SetThreshold sets the threshold for resource updates to trigger an update of the ResourceOffers.
 func (u *OfferUpdater) SetThreshold(updateThresholdPercentage uint) {
 	u.updateThresholdPercentage = updateThresholdPercentage
-	u.NotifyChange()
+	u.NotifyChange(resourcemonitors.AllClusterIDs)
 }
 
-// isAboveThreshold checks if the resources have changed by at least updateThresholdPercentage since the last update.
-func (u *OfferUpdater) isAboveThreshold(clusterID string) bool {
+// shouldUpdate checks if the resources have changed by at least updateThresholdPercentage since the last update.
+// checks are skipped if u.updateThresholdPercentage is 0.
+func (u *OfferUpdater) shouldUpdate(clusterID string) bool {
+	if u.updateThresholdPercentage == 0 {
+		return true
+	}
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
 	oldResources := u.currentResources[clusterID]
-	newResources := u.ResourceReader.ReadResources(ctx, clusterID)
-	cancel()
+	newResources, err := u.ResourceReader.ReadResources(ctx, clusterID)
+	// TODO: check if it makes sense
```

Comment:
To me, it makes sense to enqueue if an error occurs, otherwise the update would be lost. I would just add a comment explaining this fact.